src/components/ItemsSection.tsx

```diff
@@ -54,41 +72,51 @@ const ItemsSection = ({ title, itemsData }) => {
 
       <div>
         <h3>{title}</h3>
+
         <ul>
-          {items?.length === 0 ? (
-            <p>Nie dodałeś jeszcze żadnych przedmiotów.</p>
-          ) : (
-            items.map(({ image, name, stats, id }, index) => (
+          {items.map((item, index) => {
+            if (!item || !item.id) {
+              console.error('Nieprawidłowy przedmiot:', item);
+              return null;
+            }
+
+            const uniqueKey = item.id || `item-${index}`;
```

Comment:
To czy będzie sam index, czy item-index to niewiele zmienia wobec problemu który index generuje. Przy usunięciu itemu z listy wciąż cała lista zostanie przerenderowana bo zmienią się ich indeksy. jedyną opcją tutaj powinno być ID, z resztą powyżej sprawdzasz if'em czy jest ID, w takim razie to nigdy się nie zmaterializuje (item-index)